operator/v1alpha1/types_kueue.go

```diff
@@ -0,0 +1,304 @@
+	AppWrappers     KueueIntegrations = "workload.codeflare.dev/appwrapper"
+	Pod             KueueIntegrations = "pod"
+	Deployment      KueueIntegrations = "deployment"
+	Statefulset     KueueIntegrations = "statefulset"
+	LeaderWorkerSet KueueIntegrations = "leaderworkerset.x-k8s.io/leaderworkerset"
+)
+
+// This is the GVK for an external framework.
+// Controller runtime requires this in this format
+// for api discoverability.
+type ExternalFramework struct {
+	// group of externalFramework
+	// +kubebuilder:validation:MaxLength=256
+	// +required
+	Group string `json:"group"`
+	// resourceType of external framework
+	// this is the same as Kind in the GVK settings
+	// +required
+	// +kubebuilder:validation:MaxLength=256
+	ResourceType string `json:"resourceType"`
+	// version is the version of the api
+	// +required
+	// +kubebuilder:validation:MaxLength=256
+	Version string `json:"version"`
+}
+
+type Integrations struct {
+	// frameworks are a list of names to be enabled.
+	// Possible options:
+	//  - "batch/job"
+	//  - "kubeflow.org/mpijob"
+	//  - "ray.io/rayjob"
+	//  - "ray.io/raycluster"
+	//  - "jobset.x-k8s.io/jobset"
+	//  - "kubeflow.org/paddlejob"
+	//  - "kubeflow.org/pytorchjob"
+	//  - "kubeflow.org/tfjob"
+	//  - "kubeflow.org/xgboostjob"
+	//  - "workload.codeflare.dev/appwrapper"
+	//  - "pod"
+	//  - "deployment" (requires enabling pod integration)
+	//  - "statefulset" (requires enabling pod integration)
+	//  - "leaderworkerset.x-k8s.io/leaderworkerset" (requires enabling pod integration)
+	// +kubebuilder:validation:MaxItems=14
+	// +kubebuilder:validation:MinItems=1
+	// This is required and must have at least one element.
+	// The frameworks are jobs that Kueue will manage.
+	// +required
+	Frameworks []KueueIntegrations `json:"frameworks"`
+	// externalFrameworks are a list of GroupVersionKinds
+	// that are managed for Kueue by external controllers;
+	// the expected format is `Kind.version.group.com`.
+	// These are optional and should only be used if you have an external controller
+	// that integrations with kueue.
+	// +optional
+	// +kubebuilder:validation:MaxItems=4
+	ExternalFrameworks []ExternalFramework `json:"externalFrameworks,omitempty"`
+
+	// labelKeysToCopy is a list of label keys that should be copied from the job into the
+	// workload object. It is not required for the job to have all the labels from this
+	// list. If a job does not have some label with the given key from this list, the
+	// constructed workload object will be created without this label. In the case
+	// of creating a workload from a composable job (pod group), if multiple objects
+	// have labels with some key from the list, the values of these labels must
+	// match or otherwise the workload creation would fail. The labels are copied only
+	// during the workload creation and are not updated even if the labels of the
+	// underlying job are changed.
+	// +kubebuilder:validation:items:MaxLength=64
```

Comment:
Label keys are qualified names, therefore have a maximum length of 317 and should be validated with CEL

Reply:
Updated.